qoffer-domain/src/main/groovy/life/qbic/portal/portlet/offers/create/CreateOffer.groovy

```diff
@@ -8,5 +14,78 @@ package life.qbic.portal.portlet.offers.create
  * @since: 1.0.0
  * @author: Tobias Koch
  */
-class CreateOffer {
+class CreateOffer implements CreateOfferInput{
+
+    private CreateOfferDataSource dataSource
+    private CreateOfferOutput output
+
+    CreateOffer(CreateOfferDataSource dataSource, CreateOfferOutput output){
+        this.dataSource = dataSource
+        this.output = output
+    }
+
+    @Override
+    void createOffer(String tomatoId) {
+
+    }
+
+    @Override
+    void createNewOffer(Quotation quotation) {
+
+        //todo can't access the identifier, how do we want to add the identifier? Add a method to add the identifier/remove it from the constructor?
+        quotation.identifier = generateQuotationID(quotation.customer)
+        dataSource.saveOffer(quotation)
+    }
+
+    /**
+     * Method to generate the identifier of an offer with the project conserved part, the random part and the version
+     * @param customer which is required for the project conserved part
+     * @return
+     */
+    private static QuotationId generateQuotationID(Person customer){//do we want to have a person here? todo update the datamodellib
+        String projectConservedPart = customer.lastName.toLowerCase()
+        String randomPart = "abcd"
+        int version = 1
+
+        return new QuotationId(projectConservedPart,randomPart,version)
+    }
+
+    /**
+     * Method to calculate the price form the offer items
+     * @param items
+     * @return
+     */
+    private static double calculateOfferPrice(List<ProductItem> items, AffiliationCategory affiliationCategory){
```

Comment:
I know there used to be a discount based on sample size, but I assume this is (and can be?) handled in the ProductItem class?

Reply:
If we would do that it would no longer be an DTO if it contains logic. I am not sure if we want to do that.